[BUG] just init fails to install devcontainer CLI — npm global install permission denied

## Description

`just init` fails to install the `devcontainer` CLI on Linux because `npm install -g` requires write access to `/usr/local/lib/node_modules`, which is owned by root.

## Steps to Reproduce

1. Run `just init` on a Linux machine where `devcontainer` CLI is not installed
2. When prompted to install `devcontainer`, accept with `y`

## Expected Behavior

The `devcontainer` CLI should install successfully without requiring `sudo` or manual intervention.

## Actual Behavior

The installation fails with `EACCES: permission denied`:

```
  devcontainer (0.81.1)
  Purpose: DevContainer CLI for testing devcontainer functionality
  Command: npm install -g @devcontainers/cli@0.81.1
?  Install devcontainer? [y/N]: y
ℹ  Installing devcontainer...
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules'
✗  Failed to install devcontainer
```

## Environment

- **OS**: Ubuntu 24.04 (Linux 6.17.0-14-generic)
- **npm**: system npm with default global prefix (`/usr/local/lib/node_modules`)
- **Node**: system Node.js

## Additional Context

`@devcontainers/cli@0.81.1` is already declared as a local dependency in `package.json`. The `bats` dependency already uses the correct pattern — local `npm install` with a `node_modules/.bin/` check fallback. The `devcontainer` entry should follow the same approach instead of requiring a global install.

## Possible Solution

Change the `devcontainer` entry in `scripts/requirements.yaml` to:
- Use `npm install` (local) instead of `npm install -g` (global)
- Check `node_modules/.bin/devcontainer` in addition to `command -v devcontainer`
- Use `npx devcontainer --version` for version checking

This matches the existing `bats` pattern and avoids permission issues entirely.

## Changelog Category

Fixed

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -313,6 +313,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - **`just init` fails to install devcontainer CLI on Linux** ([#111](https://github.com/vig-os/devcontainer/issues/111))
   - `npm install -g` requires root access to `/usr/local/lib/node_modules`, causing EACCES permission denied
   - Switched to local `npm install` (package already declared in `package.json`), matching the existing `bats` pattern
+  - Updated pytest fixtures in `conftest.py` to also check `node_modules/.bin/devcontainer`
 
 ### Security
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -310,6 +310,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - **BATS test failures in init-workspace and prepare-build suites** ([#67](https://github.com/vig-os/devcontainer/issues/67))
   - Removed premature init-workspace.bats tests for unimplemented `is_git_dirty` feature (9 tests)
   - Fixed prepare-build.bats grep pattern for `sync_manifest.py` invocation to handle shell quoting
+- **`just init` fails to install devcontainer CLI on Linux** ([#111](https://github.com/vig-os/devcontainer/issues/111))
+  - `npm install -g` requires root access to `/usr/local/lib/node_modules`, causing EACCES permission denied
+  - Switched to local `npm install` (package already declared in `package.json`), matching the existing `bats` pattern
 
 ### Security
 

--- a/scripts/requirements.yaml
+++ b/scripts/requirements.yaml
@@ -139,17 +139,17 @@ dependencies:
       all: npm install
       manual: https://bats-core.readthedocs.io/en/stable/installation.html#any-os-npm
 
-  # DevContainer CLI (auto-installed by setup.sh)
+  # DevContainer CLI (auto-installed by npm install)
   - name: devcontainer
     version: "0.81.1"
     purpose: DevContainer CLI for testing devcontainer functionality
     required: true
     check:
-      command: command -v devcontainer
-      version_command: devcontainer --version
+      command: command -v devcontainer || test -x node_modules/.bin/devcontainer
+      version_command: npx devcontainer --version
       version_regex: '(\d+\.\d+\.\d+)'
     install:
-      all: npm install -g @devcontainers/cli@{{version}}
+      all: npm install
       manual: https://github.com/devcontainers/cli
 
   # Parallel (auto-installed by npm install)

--- a/tests/bats/init.bats
+++ b/tests/bats/init.bats
@@ -381,3 +381,15 @@ setup() {
     run grep "NC=" "$INIT_SH"
     assert_success
 }
+
+# ── devcontainer local install ───────────────────────────────────────────────
+
+@test "requirements.yaml devcontainer check falls back to node_modules/.bin" {
+    run grep 'node_modules/.bin/devcontainer' "$REQUIREMENTS_YAML"
+    assert_success
+}
+
+@test "requirements.yaml devcontainer does not use npm install -g" {
+    run grep 'npm install -g.*devcontainer' "$REQUIREMENTS_YAML"
+    assert_failure
+}

--- a/tests/bats/init.bats
+++ b/tests/bats/init.bats
@@ -393,3 +393,13 @@ setup() {
     run grep 'npm install -g.*devcontainer' "$REQUIREMENTS_YAML"
     assert_failure
 }
+
+@test "conftest.py devcontainer check falls back to node_modules/.bin" {
+    run grep 'node_modules/.bin/devcontainer' "$PROJECT_ROOT/tests/conftest.py"
+    assert_success
+}
+
+@test "conftest.py devcontainer skip message does not reference npm install -g" {
+    run grep 'npm install -g.*devcontainer' "$PROJECT_ROOT/tests/conftest.py"
+    assert_failure
+}

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -802,12 +802,24 @@ def _ensure_project_yaml_test_mount(project_config, workspace_path, in_container
     return project_config
 
 
+def _find_devcontainer_cli():
+    """Resolve the devcontainer CLI binary, checking PATH then node_modules/.bin/."""
+    path_bin = shutil.which("devcontainer")
+    if path_bin:
+        return path_bin
+    local_bin = Path("node_modules/.bin/devcontainer")
+    if local_bin.is_file():
+        return str(local_bin.resolve())
+    return None
+
+
 def _run_devcontainer_up(
     workspace_path_for_cli, workspace_path, env, docker_path="podman"
 ):
     """Run devcontainer up. Returns subprocess.CompletedProcess."""
+    devcontainer_bin = _find_devcontainer_cli() or "devcontainer"
     up_cmd = [
-        "devcontainer",
+        devcontainer_bin,
         "up",
         "--workspace-folder",
         str(workspace_path_for_cli),
@@ -881,10 +893,11 @@ def devcontainer_up(initialized_workspace):
     workspace_path, workspace_path_for_cli, in_container = (
         _resolve_devcontainer_cli_workspace(initialized_workspace)
     )
-    if not shutil.which("devcontainer") and not (
-        Path("node_modules/.bin/devcontainer").is_file()
-    ):
+    if not _find_devcontainer_cli():
         pytest.skip("devcontainer CLI not available. Install with: npm install")
+    bin_dir = str(Path("node_modules/.bin").resolve())
+    if bin_dir not in os.environ.get("PATH", ""):
+        os.environ["PATH"] = bin_dir + os.pathsep + os.environ.get("PATH", "")
 
     docker_path = "podman"
     env, original_config = _prepare_devcontainer_env(
@@ -994,9 +1007,7 @@ def devcontainer_with_sidecar(initialized_workspace, sidecar_image):
     workspace_path, workspace_path_for_cli, in_container = (
         _resolve_devcontainer_cli_workspace(initialized_workspace)
     )
-    if not shutil.which("devcontainer") and not (
-        Path("node_modules/.bin/devcontainer").is_file()
-    ):
+    if not _find_devcontainer_cli():
         pytest.skip("devcontainer CLI not available. Install with: npm install")
 
     docker_path = "podman"

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -881,11 +881,10 @@ def devcontainer_up(initialized_workspace):
     workspace_path, workspace_path_for_cli, in_container = (
         _resolve_devcontainer_cli_workspace(initialized_workspace)
     )
-    if not shutil.which("devcontainer"):
-        pytest.skip(
-            "devcontainer CLI not available. "
-            "Install with: npm install -g @devcontainers/cli"
-        )
+    if not shutil.which("devcontainer") and not (
+        Path("node_modules/.bin/devcontainer").is_file()
+    ):
+        pytest.skip("devcontainer CLI not available. Install with: npm install")
 
     docker_path = "podman"
     env, original_config = _prepare_devcontainer_env(
@@ -995,11 +994,10 @@ def devcontainer_with_sidecar(initialized_workspace, sidecar_image):
     workspace_path, workspace_path_for_cli, in_container = (
         _resolve_devcontainer_cli_workspace(initialized_workspace)
     )
-    if not shutil.which("devcontainer"):
-        pytest.skip(
-            "devcontainer CLI not available. "
-            "Install with: npm install -g @devcontainers/cli"
-        )
+    if not shutil.which("devcontainer") and not (
+        Path("node_modules/.bin/devcontainer").is_file()
+    ):
+        pytest.skip("devcontainer CLI not available. Install with: npm install")
 
     docker_path = "podman"
     env, _ = _prepare_devcontainer_env(

